internal/clients/config/elasticsearch.go

```diff
@@ -38,6 +38,14 @@ func newElasticsearchConfigFromSDK(d *schema.ResourceData, base baseConfig, key
 			config.Addresses = addrs
 		}
 
+		if bearer_token, ok := esConfig["bearer_token"].(string); ok && bearer_token != "" {
+			base.Header.Set("Authorization", bearer_token)
+		}
+
+		if es_client_authentication, ok := esConfig["es_client_authentication"].(string); ok && es_client_authentication != "" {
+			base.Header.Set("ES-Client-Authentication", es_client_authentication)
+		}
```

Comment:
This will work for ES (unfortunately, it kind of exposes a bug in how we deal with this header collection TBH), but is likely to break Kibana/Fleet clients. 

We should:
* Give the ES client it's own copy of the base headers `b.Header.Clone()` [here](https://github.com/elastic/terraform-provider-elasticstack/blob/main/internal/clients/config/base.go#L80)
* Set these headers on the ES config rather than the base config. 

```suggestion
		if bearer_token, ok := esConfig["bearer_token"].(string); ok && bearer_token != "" {
			config.Header.Set("Authorization", bearer_token)
		}

		if es_client_authentication, ok := esConfig["es_client_authentication"].(string); ok && es_client_authentication != "" {
			config.Header.Set("ES-Client-Authentication", es_client_authentication)
		}
```